sklearn/calibration.py

```diff
@@ -753,3 +755,209 @@ def calibration_curve(y_true, y_prob, *, normalize=False, n_bins=5,
+
+        existing_ref_line = ('Perfectly calibrated' in
+                             ax.get_legend_handles_labels()[1])
+        if ref_line and not existing_ref_line:
+            ax.plot([0, 1], [0, 1], "k:", label="Perfectly calibrated")
+        self.line_ = ax.plot(self.prob_pred, self.prob_true, "s-",
+                             **line_kwargs)[0]
+
+        if "label" in line_kwargs:
+            ax.legend(loc="lower right")
+
+        ax.set(xlabel="Mean predicted probability",
+               ylabel="Fraction of positives")
+
+        self.ax_ = ax
+        self.figure_ = ax.figure
+        return self
+
+
+def plot_calibration_curve(estimator, X, y, *,
+                           n_bins=5, strategy='uniform',
+                           name=None, ref_line=True, ax=None, **kwargs):
+    """Plot calibration curve for binary classifiers.
+
+    The average predicted probability for each bin is plotted on the x-axis
+    and the fraction of positive classes in each bin is plotted on the y-axis.
+
+    Extra keyword arguments will be passed to :func:`matplotlib.pyplot.plot`.
+
+    Read more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    ----------
+    estimator : estimator instance
+        Fitted classifier or a fitted :class:`~sklearn.pipeline.Pipeline` in
+        which the last estimator is a classifier. The classifier must
+        have a :term:`predict_proba` method; set `probability=True` for
+        :mod:`~sklearn.svm` estimators (see:
+        :ref:`User Guide <scores_probabilities>`) or use
+        :class:`~sklearn.calibration.CalibratedClassifierCV`.
+
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        Input values.
+
+    y : array-like of shape (n_samples,)
+        Binary target values.
+
+    n_bins : int, default=5
+        Number of bins to discretize the [0, 1] interval into when calculating
+        the calibration curve. A bigger number requires more data.
+
+    strategy : {'uniform', 'quantile'}, default='uniform'
+        Strategy used to define the widths of the bins.
+
+        - `'uniform'`: The bins have identical widths.
+        - `'quantile'`: The bins have the same number of samples and depend on
+          predicted probabilities.
+
+    name : str, default=None
+        Name for labeling curve. If `None`, the name of the estimator is used.
+
+    ref_line : bool, default=True
+        If `True`, plots a reference line representing a perfectly calibrated
+        classifier.
+
+    ax : matplotlib axes, default=None
+        Axes object to plot on. If `None`, a new figure and axes is created.
+
+    **kwargs : dict
+        Keyword arguments to be passed to :func:`matplotlib.pyplot.plot`.
+
+    Returns
+    -------
+    display : :class:`~sklearn.calibration.CalibrationDisplay`.
+        Object that stores computed values.
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt                     # doctest: +SKIP
```

Comment:
I had a look at the `conftest.py` files with `find . -name "conftest.py" | xargs grep matplotlib` and appart from the `pyplot` fixture used in the regular unit test, I do not see anything related to the use of matplotlib in doctests.

The top level `conftest,py` file also has a `pytest_collection_modifyitems` that makes it possible to skip specifc class doctests for various reasons but that seems cumbersome to maintain manually.

Maybe there is a way to setup an automatic thing to introspect the text of the class and to check if "matplotlib" is mentioned, in which case the `pytest_collection_modifyitems` would mark the doctest of the class to be skipped if matplotlib is not installed or configure the "agg" headless backend otherwise (or always skip even if matplotlib is installed).

Reply:
it seems we're importing pyplot in `plotting.rst` so I think we should remove the doctest skip. As seen below, this hides bugs